sdcm/sct_events.py

```diff
@@ -0,0 +1,446 @@
+            )
+
+
+class CassandraStressEvent(SctEvent):
+    def __init__(self, type, node, stress_cmd=None, log_file_name=None, errors=None):
+        super(CassandraStressEvent, self).__init__()
+        self.type = type
+        self.node = str(node)
+        self.stress_cmd = stress_cmd
+        self.log_file_name = log_file_name
+        self.priority = Severity.NORMAL
+        self.errors = errors
+        self.publish()
+
+    def __str__(self):
+        if self.stress_cmd:
+            return "{}: type={} node={}\nstress_cmd={}".format(
+                super(CassandraStressEvent, self).__str__(), self.type, self.node, self.stress_cmd)
+        if self.errors:
+            return "{}: type={} node={}\n{}".format(
+                super(CassandraStressEvent, self).__str__(), self.type, self.node, "\n".join(self.errors))
+
+
+class ScyllaLogEvent(SctEvent):
+    def __init__(self, name, regex, severity=Severity.CRITICAL):
+        super(ScyllaLogEvent, self).__init__()
+        self.name = name
+        self.regex = regex
+        self.line_no = 0
+        self.line = None
+        self.node = None
+        self.severity = severity
+
+    def add_info_and_publish(self, node, line, line_no):
+        self.timestamp = self.set_timestamp()
+        self.line = line
+        self.line_no = line_no
+        self.node = str(node)
+        super(ScyllaLogEvent, self).publish()
+
+    def __str__(self):
+        return "{}: name={} node={} line_no={} regex={}\n{}".format(
+            super(ScyllaLogEvent, self).__str__(), self.name, self.node, self.line_no, self.regex, self.line)
+
+
+class SpotTerminationEvent(SctEvent):
+    def __init__(self, node, action):
+        super(SpotTerminationEvent, self).__init__()
+        self.severity = Severity.CRITICAL
+        self.node = str(node)
+        self.action = action
+        self.publish()
+
+    def __str__(self):
+        return "{}: node={}  action={} ".format(
+            super(SpotTerminationEvent, self).__str__(), self.node, self.action)
+
+
+def pid_exists(pid):
+    """
+    Return True if a given PID exists.
+
+    :param pid: Process ID number.
+    """
+    try:
+        os.kill(pid, 0)
+    except OSError as detail:
+        if detail.errno == errno.ESRCH:
+            return False
+    return True
+
+
+def safe_kill(pid, signal):
+    """
+    Attempt to send a signal to a given process that may or may not exist.
+
+    :param signal: Signal number.
+    """
+    try:
+        os.kill(pid, signal)
+        return True
+    except Exception:
+        return False
+
+
+class TestKiller(Process):
+    def __init__(self, timeout_before_kill=2, test_callback=None):
+        super(TestKiller, self).__init__()
+        self._test_pid = os.getpid()
+        self.test_callback = test_callback
+        self.timeout_before_kill = timeout_before_kill
+
+    def run(self):
+        for event_type, message_data in EVENTS_PROCESS.subscribe_events():
+            if event_type == 'KillTestEvent':
```

Comment:
should be SystemEvent

Reply:
right now I would filter SystemEvents from subscribe, but would Like to implement the TestKiller as subscriber, so I don't think I'll change that event into system event